app/jobs/update_job.rb

```diff
@@ -50,7 +50,10 @@ def perform(model_params:,
       return
     end
 
-    # Note that not using a lock here since all model params are being provided rather than updating retrieved params.
+    # globus deposits may not have digests yet and they need to be generated before staging (copy)
+    model = GlobusDigestGenerator.generate(cocina: model, globus_ids: globus_ids)
+
+    # not using a lock here since all model params are being provided rather than updating retrieved params.
     object_client.update(params: model, skip_lock: true)
```

Comment:
I wasn't sure if `skip_lock` should be set to `false` now that it's possible the model params will have been updated?

Reply:
I'm not sure. Maybe @justinlittman has a sense?

Reply:
`skip_lock` should be false since this is always a regeneration of the cocina (even if part of it is generated by SDR API).

`skip_lock` only makes sense when the cocina is being retrieved, modified, and then replaced.

Reply:
@justinlittman it might not have been clear from the snippet but I believe the Cocina is being fetched on line 34 and then the call to `GlobusDigestGenerator.generate` is adding new digests to the object (if they are missing), and then the Cocina is being saved again?

Reply:
The line numbering is off, so I'm going to snippet below and add annotations about what I think is happening:
```
# Model is being built from scratch from provided params.
model = Cocina::Models.build(model_params.with_indifferent_access)

# Existing is being retrieved, but only used to compare versions.
object_client = Dor::Services::Client.object(model.externalIdentifier)
existing = object_client.find

.....
# The built model is being updated.
model = GlobusDigestGenerator.generate(cocina: model, globus_ids: globus_ids)
```
If my reading is correct, then I still think that we want `skip_lock`.

Reply:
After chatting about this with @justinlittman we are continuing to go with `skip_lock: true` as it stands, since `model` comes from H2 and cannot be used for locking.